frontend/src/component/releases/ReleasePlanTemplate/EditReleasePlanTemplate.tsx

```diff
@@ -0,0 +1,139 @@
+
+const StyledForm = styled('form')(() => ({
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+}));
+
+const StyledMilestoneCard = styled(Card)(({ theme }) => ({
+    marginTop: theme.spacing(2),
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    boxShadow: 'none',
+    border: `1px solid ${theme.palette.divider}`,
+    [theme.breakpoints.down('sm')]: {
+        justifyContent: 'center',
+    },
+    transition: 'background-color 0.2s ease-in-out',
+    backgroundColor: theme.palette.background.default,
+    '&:hover': {
+        backgroundColor: theme.palette.neutral.light,
+    },
+    borderRadius: theme.shape.borderRadiusMedium,
+}));
+
+const StyledMilestoneCardBody = styled(Box)(({ theme }) => ({
+    padding: theme.spacing(3, 2, 3, 2),
+}));
+
+const StyledMilestoneCardTitle = styled('span')(({ theme }) => ({
+    fontWeight: theme.fontWeight.bold,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+const StyledButtonContainer = styled('div')(() => ({
+    marginTop: 'auto',
+    display: 'flex',
+    justifyContent: 'flex-end',
+}));
+
+const StyledCancelButton = styled(Button)(({ theme }) => ({
+    marginLeft: theme.spacing(3),
+}));
+
+export const EditReleasePlanTemplate = () => {
+    const releasePlansEnabled = useUiFlag('releasePlans');
+    if (!releasePlansEnabled) {
+        return null;
+    }
+
+    const templateId = useRequiredPathParam('templateId');
+    const { template, loading, error, refetch } =
+        useReleasePlanTemplateInstance(templateId);
+    usePageTitle(`Edit template: ${template.name}`);
+    const navigate = useNavigate();
+    const { setToastApiError } = useToast();
+    const { updateReleasePlanTemplate } = useReleasePlanTemplatesApi();
+    const {
+        name,
+        setName,
+        description,
+        setDescription,
+        errors,
+        clearErrors,
+        validate,
+        getTemplatePayload,
+    } = useTemplateForm(template.name, template.description);
+
+    const handleCancel = () => {};
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        clearErrors();
+        const isValid = validate();
+        if (isValid) {
+            const payload = getTemplatePayload();
+            try {
+                updateReleasePlanTemplate({
+                    ...payload,
+                    id: templateId,
+                    milestones: template.milestones,
+                }).then(() => {
+                    navigate('/release-management');
+                });
+            } catch (error: unknown) {
+                setToastApiError(formatUnknownError(error));
+            }
+        }
+    };
+
+    return (
+        <>
+            <FormTemplate
+                title={`Edit template ${template.name}`}
+                description='Edit a release plan template that makes it easier for you and your team to release features.'
+            >
```

Comment:
I assume we'll add other props like `formatApiCode` at a later stage?

Reply:
Yeah, we have to look at extending this to fit other type of information as well